promptsource/seqio_tasks/tasks.py

```diff
@@ -1,44 +1,38 @@
+import csv
 import functools
 import re
+from typing import Dict, List, Optional, Tuple
 
 import datasets
+import pkg_resources
 import seqio
 import t5
 import tensorflow as tf
+from t5.data.glue_utils import get_glue_metric, get_super_glue_metric
+from t5.evaluation import metrics as mt
 
 import promptsource.templates
-
-from . import load_annotated_prompts, utils
-
-
-# Tasks deemed as clean/useful
-annotated_tasks = load_annotated_prompts.load_annotated_prompts()
-CLEAN_TASKS = [t["dataset_subset_template"] for t in annotated_tasks if not t["skip_train"]]
-CLEAN_EVAL_TASKS = [t["dataset_subset_template"] for t in annotated_tasks if t["do_eval"]]
-EVAL_METRICS = {t["dataset_subset_template"]: t["metrics"] for t in annotated_tasks if t["do_eval"]}
-
-
-# Datasets that don't work currently...
-DATASET_BLACKLIST = [
-    ("species_800", None),
-    ("drop", None),
-    ("discofuse", "discofuse-sport"),
-    ("discofuse", "discofuse-wikipedia"),
-    ("adversarial_qa", "adversarialQA"),
-    ("tweet_eval", "emotion"),
-    ("tweet_eval", "emoji"),
-    ("tweet_eval", "hate"),
-    ("tweet_eval", "offensive"),
-    ("tweet_eval", "stance_atheism"),
-    ("tweet_eval", "stance_abortion"),
-    ("tweet_eval", "stance_feminist"),
-    ("tweet_eval", "stance_climate"),
-    ("tweet_eval", "sentiment"),
-    ("tweet_eval", "stance_hillary"),
-    ("tweet_eval", "irony"),
-    # Need to special-case ANLI due to weird split conventions
-    ("anli", None),
-]
+from promptsource.seqio_tasks import utils
+
+
+GET_METRICS = {
+    "BLEU": mt.bleu,
+    "ROUGE": mt.rouge,
+    "Span Squad": mt.span_squad,
+    "Squad": mt.squad,
+    "Trivia QA": mt.trivia_qa,
+    "Accuracy": mt.accuracy,
+    "Sequence Accuracy": mt.sequence_accuracy,
+    "Pearson Correlation": mt.pearson_corrcoef,
+    "Spearman Correlation": mt.spearman_corrcoef,
+    "MultiRC": mt.multirc_f1_over_all_answers,
+    "AUC": mt.auc,
+    "COQA F1": mt.coqa_f1,
+    "Edit Distance": mt.edit_distance,
+    # "Mean Reciprocal Rank": mt.accuracy,  # NOTE not in T5?
```

Comment:
Which datasets use mrr?

Reply:
I think none so far. But @stephenbach added it as an option [here](https://github.com/bigscience-workshop/promptsource/blob/26841476fc58682e9749bbd5cf77fb2f98129f22/promptsource/promptsource.py#L509), so I thought I'd put it here as a commented out placeholder.